pkg/ddl-checker/ddl_checker_test.go

```diff
@@ -0,0 +1,93 @@
+	parseSucceeded      bool
+	tableNeededExist    []string
+	tableNeededNonExist []string
+	executeSucceeded    bool
+}
+
+func TestT(t *testing.T) {
+	TestingT(t)
+}
+func (s *testSuite) SetUpSuite(c *C) {
+	var err error
+	s.ec, err = NewExecutableChecker()
+	c.Assert(err, IsNil)
+	s.setUpTestData()
+}
+
+func (s *testSuite) setUpTestData() {
+	s.testData = list.New()
+
+	s.testData.PushBack(parseTestData{sql: "drop table if exists t1,t2,t3,t4,t5;", parseSucceeded: true, tableNeededExist: []string{"t1", "t2", "t3", "t4", "t5"}, tableNeededNonExist: []string{}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "drop database if exists mysqltest;", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "create table t1 (b char(0));", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{"t1"}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "insert into t1 values (''),(null);", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "select * from t1;", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "drop table if exists t1;", parseSucceeded: true, tableNeededExist: []string{"t1"}, tableNeededNonExist: []string{}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "create table t1 (b char(0) not null);", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{"t1"}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "create table if not exists t1 (b char(0) not null);", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{"t1"}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "insert into t1 values (''),(null);", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{}, executeSucceeded: false})
+	s.testData.PushBack(parseTestData{sql: "select * from t1;", parseSucceeded: true, tableNeededExist: []string{}, tableNeededNonExist: []string{}, executeSucceeded: true})
+	s.testData.PushBack(parseTestData{sql: "drop table t1;", parseSucceeded: true, tableNeededExist: []string{"t1"}, tableNeededNonExist: []string{}, executeSucceeded: true})
+
+}
+
+func (s *testSuite) TearDownSuite(c *C) {
+	s.ec.Close()
+}
+func (s *testSuite) TestParse(c *C) {
+	for e := s.testData.Front(); e != nil; e = e.Next() {
+		data := e.Value.(parseTestData)
+		stmt, err := s.ec.Parse(data.sql)
+		if err != nil {
+			c.Assert(data.parseSucceeded, IsFalse)
+			continue
+		}
+		tableNeededExist := GetTableNeededExist(stmt)
+		tableNeededNonExist := GetTableNeededNonExist(stmt)
+		c.Assert(data.parseSucceeded, IsTrue)
+		c.Assert(reflect.DeepEqual(data.tableNeededExist, tableNeededExist), IsTrue)
```

Comment:
```suggestion
		c.Assert(data.tableNeededExist, DeepEquals, tableNeededExist)
```

Similar to the rest.